misc/ballerina-format/src/main/java/org/ballerinalang/format/FormatUtil.java

```diff
@@ -0,0 +1,387 @@
+                        }
+                        BLangPackage bLangPackage = FormatUtil
+                                .compileModule(sourceRootPath, moduleName);
+
+                        // If there is no compilation error continue the process.
+                        if (!bLangPackage.diagCollector.hasErrors()) {
+                            if (overWriteFile) {
+                                iterateAndFormat(bLangPackage, sourceRootPath);
+                            } else {
+                                outStream.println(Messages.getConfirmationMessage());
+                                Scanner userInput = new Scanner(System.in, "UTF-8");
+                                boolean wait = true;
+                                while (wait) {
+                                    String input = userInput.nextLine();
+                                    if (input != null) {
+                                        if (input.equals("y")) {
+                                            iterateAndFormat(bLangPackage, sourceRootPath);
+                                            wait = false;
+                                        } else {
+                                            outStream.println(Messages.getCancelMessage());
+                                            wait = false;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    } else {
+                        // If module directory doesn't exist and contains a "."
+                        // throw a exception to say file or module doesn't exist.
+                        // Else throw a exception to say module doesn't exist.
+                        if (moduleName.contains(".")) {
+                            throw LauncherUtils.createLauncherException(Messages
+                                    .getNoBallerinaModuleOrFile(moduleName));
+                        } else {
+                            throw LauncherUtils.createLauncherException(Messages.getNoModuleFound(moduleName));
+                        }
+                    }
+                }
+            } else {
+                // Check whether the given source root is not
+                if (FormatUtil.notABallerinaProject(sourceRootPath)) {
+                    throw LauncherUtils.createLauncherException(Messages.getNotBallerinaProject());
+                }
+
+                List<BLangPackage> packages = FormatUtil.compileProject(sourceRootPath);
+                boolean noCompilationError = true;
+                for (BLangPackage bLangPackage : packages) {
+                    if (bLangPackage.diagCollector.hasErrors()) {
+                        noCompilationError = false;
+                        break;
+                    }
+                }
+                if (noCompilationError) {
```

Comment:
Inversing the `if` condition will reduce the number of nested blocks.

```java
if (!noCompilationError) { // has compilation errors :)
    return;
}
```

So I think it would be better to rename the variable to `hasCompilationErrors` to increase the readability ..

Reply:
will fix